src/meta/src/barrier/progress.rs

```diff
@@ -107,6 +108,26 @@ impl<S: MetaStore> CreateMviewProgressTracker<S> {
         }
     }
 
+    /// Try to cancel a target create-mview command from track.
+    ///
+    /// If the actors to cancel is not empty, return the target command as it should be cancelled.
+    pub fn cancel(&mut self, context: &Arc<CommandContext<S>>) -> Option<TrackingCommand<S>> {
```

Comment:
This function seems a little bit confusing to me, as if we're always canceling the current command. What about moving the `if !actors_to_cancel.is_empty()` to the barrier manager?